### PR TITLE
Internal 824 kafka e2e test

### PR DESCRIPTION
**Pull request by jgrillo-grapl**
### Which issue does this PR correspond to?
https://github.com/grapl-security/issue-tracker/issues/824

### What changes does this PR make to Grapl? Why?
This PR adds code and configuration to run an end-to-end test which exercises kafka locally. These changes also remove the `kafka-metrics-exporter` which is no longer used.

### How were these changes tested?
`make test-e2e` locally.

### CODE REVIEW

File: src/python/e2e-test-runner/e2e_test_runner/test_kafka.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import copy
+import os
+import uuid
+from typing import Any
+
+import pytest
+from confluent_kafka import Consumer, Producer  # type: ignore
+
+IS_LOCAL = not os.environ["KAFKA_BOOTSTRAP_SERVERS"].startswith("SASL_SSL")
```
Comment (wimax-grapl) [inline src/python/e2e-test-runner/e2e_test_runner/test_kafka.py:9]: I know i told you to do this, but I'm not sure I love this. Hmm. Well, it's fine for now. I guess all this comment is saying is "huh maybe we should revisit in the future or something"

File: src/python/e2e-test-runner/e2e_test_runner/test_kafka.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    yield consumer
+    consumer.close()
+
+
+def _producer_callback(err: Any, _: Any) -> None:
+    assert err is None
+
+
+def test_kafka_can_write_metrics(
+    kafka_producer: Producer, metrics_consumer: Consumer
```
Comment (wimax-grapl) [inline src/python/e2e-test-runner/e2e_test_runner/test_kafka.py:67]: nice use of test fixtures 🙏

**Review by wimax-grapl — APPROVED** [review]

File: pulumi/grapl/__main__.py [hunk 3]
```diff
@@ -234,7 +243,7 @@ def main() -> None:
         redis_endpoint = "redis://LOCAL_GRAPL_REPLACE_IP:6379"
 
         pulumi.export("aws-endpoint", aws_endpoint)
-        pulumi.export("kafka-endpoint", kafka.bootstrap_servers())
+        pulumi.export("kafka-bootstrap-servers", kafka.bootstrap_servers())
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply pulumi/grapl/__main__.py]: fixed

File: nomad/e2e-tests.nomad [hunk 4]
```diff
@@ -203,7 +224,14 @@ EOF
         RUST_BACKTRACE = 1
         RUST_LOG       = local.log_level
 
+        KAFKA_BOOTSTRAP_SERVERS = local.kafka_bootstrap_servers
+        KAFKA_SASL_USERNAME     = var.kafka_sasl_username
+        KAFKA_SASL_PASSWORD     = var.kafka_sasl_password
+      }
+
+      resources {
+        memory = 1024
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply nomad/e2e-tests.nomad]: removed

File: nomad/local/stop_detach.sh [hunk 5]
```diff
@@ -4,3 +4,6 @@ set -euo pipefail
 
 sudo killall --wait nomad || true
 killall --wait consul || true
+docker volume rm local-grapl-zk-data --force
+docker volume rm local-grapl-zk-txn-logs --force
+docker volume rm local-grapl-kafka-data --force
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply nomad/local/stop_detach.sh]: this was actually left over from experiments with named volumes and isn't necessary, removed

File: pulumi/infra/kafka.py [hunk 6]
```diff
@@ -116,6 +118,42 @@ def __init__(
 
         if confluent_environment_name == "local-grapl":
             self.confluent_environment = None
+            # This list must match the set of all topics specified in
+            # confluent-cloud-infrastructure/pulumi/ccloud_bootstrap/index.ts
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply pulumi/infra/kafka.py:122]: yes, will open PR

**Pull request merged**